workflow/controller/operator.go

```diff
@@ -2108,6 +2107,43 @@ func (woc *wfOperationCtx) onNodeComplete(node *wfv1.NodeStatus) {
 	)
 }
 
+// recordNodePhaseChangeEvents creates WorkflowNode Kubernetes events for each node
+// that has changes logged during this execution of the operator loop.
+func (woc *wfOperationCtx) recordNodePhaseChangeEvents(old *wfv1.Nodes, new *wfv1.Nodes) {
+	if !woc.controller.Config.NodeEvents.IsEnabled() {
+		return
+	}
+	oldNodes := *old
```

Comment:
I think you can remove these pointer dereferences. I think the code works without them.

Reply:
I needed to retain the pointer de-references, but I removed the extra variable assignment for the new nodes.

Reply:
https://github.com/argoproj/argo-workflows/pull/5531/commits/fd20545f4effc6f1174df55ffb6b62127b0b1730